tests/foreman/cli/test_architecture.py

```diff
@@ -33,27 +24,22 @@ def test_negative_update(self, data):
         @assert: architecture name is not updated
 
         """
-        try:
-            new_obj = make_architecture()
-        except CLIFactoryError as err:
-            self.fail(err)
+        for data in invalid_values_list():
+            with self.subTest(data):
+                try:
+                    new_obj = make_architecture()
```

Comment:
You can move the creation outside the loop and perform only the update inside the loop to speed up the test.